src/Compilers/Core/Desktop/CommandLine/CommonCompiler.cs

```diff
@@ -527,6 +528,9 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                 if (analyzerCts != null)
                 {
                     analyzerCts.Cancel();
+
+                    // Clear cached analyzer descriptors and unregister exception handlers hooked up to the LocalizableString fields of the associated descriptors.
+                    analyzerManager.ClearCachedAnalyzerDescriptors(analyzers);
```

Comment:
Additionally, we need cleanup code for IDE as analyzer manager instance lives throughout VS instance, but per-project analyzers might come and go, so state for those analyzers needs to be cleared. We already did that though.